### DISCUSSION TITLE
Charcoal effect and transparency.

### DISCUSSION THREAD

**Discussion by artnose** [ImageMagick/ImageMagick#2997] [General] [answered]
The -charcoal effect works good for me, until part of the image is transparent, 
then, the effect completely changes.   Is there some interaction with transparency? 
How can I get the charcoal effect to work w/ transparent images.

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform?

"-charcoal" uses a number of lower-level processes: edge, clamp, blur, normalize, negate and grayscale. Yes, transparency has an effect on most of these.

You can copy the opacity, switch it off, do the charcoal effect, and copy the opacity back. Windows format::
```
magick ^
  in.png ^
  ( +clone -alpha extract -write mpr:ALPH +delete) ^
  -alpha off -charcoal 0x1 mpr:ALPH -compose CopyOpacity -composite ^
  out.png
```
... but if you have a change of colour that was hidden by transparency, this will give a charcoal effect there.

**Comment by artnose** [comment]
Ok, thank you, that works great!
